modules/core/src/lib/deck.js

```diff
@@ -436,32 +441,55 @@ export default class Deck {
     return views;
   }
 
-  _pickAndCallback(options) {
-    const pos = options.event.offsetCenter;
-    // Do not trigger callbacks when click/hover position is invalid. Doing so will cause a
-    // assertion error when attempting to unproject the position.
-    if (!pos) {
-      return;
+  _requestPick({event, callback, mode, immediate}) {
```

Comment:
Add comment headers to these methods explain the intended function so that future maintainers understand why this was done.